File: Arrays_Strings/python/urlify.py
def urlify(string, length):
    spaces = 0

    print(string)

    for s in string[:length]:
        if s == " ":
            spaces += 1

    index = length + spaces * 2

    i = length - 1

    for s in string[:length][::-1]:
        if s != " ":
            string[index - 1] = string[i]
            index -= 1
        else:
            string[index - 1] = "0"
            string[index - 2] = "2"
            string[index - 3] = "%"
            index -= 3
        i -= 1

    return ''.join(string)


def main():
    string = "Mr John Smith    "
    print(urlify(list(string), 13))


if __name__ == '__main__':
    main()

File: Arrays_Strings/go/urlify.go
package main
import (
  "fmt"
)


func urlify(s []byte, length int) string {
  spaces := 0
  for i := 0; i < length; i++ {
    if string(s[i]) == " "{
      spaces++
    }
  }

  index := length + spaces * 2
  for i := length - 1; i >= 0; i-- {
    if string(s[i]) != " " {
        s[index - 1] = s[i]
        index--

    } else {
      s[index - 1] = '0'
      s[index - 2] = '2'
      s[index - 3] = '%'

      index -= 3
    }
  }

  return string(s)
}


func main() {
  var str []byte = []byte("Mr John Smith    ")

  fmt.Println(urlify(str, 13))
}
